logstash-core/src/main/java/org/logstash/instrument/metrics/counter/CounterMetric.java

```diff
@@ -0,0 +1,22 @@
+package org.logstash.instrument.metrics.counter;
+
+import org.logstash.instrument.metrics.Metric;
+
+/**
+ * A {@link Metric} to count things. A counter can only increment, and there are no guarantees of durability of current value between JVM restarts.
+ * @param <T> The underlying {@link Number} type that can be incremented. Care should be taken to which {@link Number} is used to back the counter, since some {@link Number}
+ *           types may not be appropriate for a monotonic increasing series.
+ */
+public interface CounterMetric<T extends Number> extends Metric<T> {
```

Comment:
For consistency for the Java API's I think it still makes sense. This would be available to plugin developers to implement other type of counters such BigInteger if needed. It also allows for a passive path for additions of other types of counters.